packages/@uppy/dashboard/src/components/FileCard/index.jsx

```diff
@@ -26,8 +26,7 @@ class FileCard extends Component {
     this.form.id = nanoid()
   }
 
-  // TODO(aduh95): move this to `UNSAFE_componentWillMount` when updating to Preact X+.
-  componentWillMount () { // eslint-disable-line react/no-deprecated
+  componentDidMount () { // eslint-disable-line react/no-deprecated
```

Comment:
By moving that to `componentDidMount`, we technically add a race condition if the submit button happens right between the `render` step and `componentDidMount`. Any reason not to use `UNSAFE_componentWillMount`?

Reply:
The word `UNSAFE` scared me off and I decided to play it safe.

Reply:
can this race condition even happen in real life? is the browser even interactive before initial useEffects get called? I'm thinking it's not theoretically possible to press the submit button before useEffect has been called because I think it all happens in the same tick, but i don't have any evidence for that.

Alternatively a rewrite to hooks could allow us to keep it theoretically backward compatible, and it would allow us to use snacks like `useId`, so that we maybe one day could remove nanoid as a dependency.

Reply:
> I think it all happens in the same tick, but i don't have any evidence for that

It's actually quite trivial to test that:

```js
class MyComponent extends Component {
  UNSAFE_componentWillMount() {
    queueMicrotask(() => alert('UNSAFE_componentWillMount next tick'));
  }

  componentDidMount() {
    alert('componentDidMount tick');
  }

  render() { return <div/>; }
}
```

Just tried it in the [Preact REPL](https://preactjs.com/repl) and `componentDidMount tick` displays first, showing that it either happens in the same tick indeed, or at least the `componentDidMount` is already scheduled.
I wonder how it affects browser perf to add `<input>` elements with an unknown `form` ID, and then add the `<form>` to the DOM.

Reply:
Thanks for the feedback! How does a rewrite to hooks help keep this backward-compatible, what exactly should be compatible?

Reply:
Gives us control to run the code immediately on render. for example `useState(() => document.createElement...)`